### PR TITLE
bpo-42918: Improve build-in function compile() in mode 'single'

### PR DESCRIPTION
**Pull request by hongweipeng**
https://bugs.python.org/issue42918

### CODE REVIEW

File: Misc/NEWS.d/next/Core and Builtins/2021-12-06-15-32-12.bpo-42918.Czpgtg.rst [hunk 1]
```diff
@@ -0,0 +1 @@
+Improve build-in :func:`compile` in mode 'single'.  Patch by Weipeng Hong.
```
Comment (pablogsal) [inline Misc/NEWS.d/next/Core and Builtins/2021-12-06-15-32-12.bpo-42918.Czpgtg.rst]: Could you please add more context on what this PR is fixing here? This current text is unfortunately too generic.

**Review by pablogsal — APPROVED** [review]

**Comment by pablogsal** [comment]
LGTM

I can merge once the NEWS entry is fixed :)

**Comment by pablogsal** [comment]
@hongweipeng Please, ping me here once you make the changes

**Comment by hongweipeng** [comment]
@pablogsal I have made the requested changes; please review again.

File: Misc/NEWS.d/next/Core and Builtins/2021-12-06-15-32-12.bpo-42918.Czpgtg.rst [hunk 2]
```diff
@@ -0,0 +1,2 @@
+The build-in :func:`compile` should raise :exc:`SyntaxError` when it compile
+multiple statements in 'single' mode. Patch by Weipeng Hong.
```
Comment (AlexWaygood) [inline Misc/NEWS.d/next/Core and Builtins/2021-12-06-15-32-12.bpo-42918.Czpgtg.rst]:
```suggestion
Fix bug where the built-in :func:`compile` function did not always raise a
:exc:`SyntaxError` when passed multiple statements in 'single' mode. Patch by
Weipeng Hong.
```

**Pull request merged**